Subscription hook is failing linter

```
> scaffold-stellar-frontend@0.0.1 lint
> eslint .


/home/runner/work/scaffold-stellar-frontend/scaffold-stellar-frontend/src/hooks/useSubscription.ts
  43:15  error    'latestLedgerState' is never reassigned. Use 'const' instead                                                                                                        prefer-const
  47:13  error    'response' is never reassigned. Use 'const' instead                                                                                                                 prefer-const
  68:9   error    Expected an assignment or function call and instead saw an expression                                                                                               @typescript-eslint/no-unused-expressions
  81:34  error    Promise returned in function argument where a void return was expected                                                                                              @typescript-eslint/no-misused-promises
  86:5   error    Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises
  92:6   warning  React Hook React.useEffect has a missing dependency: 'server'. Either include it or remove the dependency array                                                     react-hooks/exhaustive-deps

✖ 6 problems (5 errors, 1 warning)
  2 errors and 0 warnings potentially fixable with the `--fix` option.
```

--- a/src/hooks/useSubscription.ts
+++ b/src/hooks/useSubscription.ts
@@ -40,11 +40,11 @@ export function useSubscription(
     async function pollEvents(): Promise<void> {
       try {
         if (!paging[id].lastLedgerStart) {
-          let latestLedgerState = await server.getLatestLedger();
+          const latestLedgerState = await server.getLatestLedger();
           paging[id].lastLedgerStart = latestLedgerState.sequence
         } 
        
-        let response = await server.getEvents({
+        const response = await server.getEvents({
           startLedger: !paging[id].pagingToken
             ? paging[id].lastLedgerStart
             : undefined,
@@ -60,34 +60,36 @@ export function useSubscription(
           ],
           limit: 10
         });
-     
+
         paging[id].pagingToken = undefined;
         if (response.latestLedger) {
           paging[id].lastLedgerStart = response.latestLedger;
         }
-        response.events && response.events.forEach(event => {
-          try {
-            onEvent(event)
-          } catch (error) {
-            console.error("Poll Events: subscription callback had error: ", error);
-          } finally {
-            paging[id].pagingToken = event.pagingToken
-          }
-        }) 
+        if (response.events) {
+          response.events.forEach(event => {
+            try {
+              onEvent(event)
+            } catch (error) {
+              console.error("Poll Events: subscription callback had error: ", error);
+            } finally {
+              paging[id].pagingToken = event.pagingToken
+            }
+          }) 
+        }
       } catch (error) {
         console.error("Poll Events: error: ", error);
       } finally {
         if (!stop) {
-          timeoutId = setTimeout(pollEvents, pollInterval);
+          timeoutId = setTimeout(() => void pollEvents(), pollInterval);
         }
       }
     }
 
-    pollEvents();
+    void pollEvents();
 
     return () => {
       if (timeoutId != null) clearTimeout(timeoutId)
       stop = true
     }
-  }, [contractId, topic, onEvent, id, pollInterval])
+  }, [contractId, topic, onEvent, id, pollInterval, server])
 }

